File: services/ReadingService1/src/main/java/com/se1853_jv/reading/repository/ReadingWorkflowHighlightRepository.java
package com.se1853_jv.reading.repository;

import com.se1853_jv.reading.model.ReadingWorkflowHighlight;
import com.se1853_jv.reading.model.ReadingWorkflowHighlightId;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface ReadingWorkflowHighlightRepository extends JpaRepository<ReadingWorkflowHighlight, ReadingWorkflowHighlightId> {

    List<ReadingWorkflowHighlight> findByIdCollectionIdAndIdPaperIdAndIdUserId(String collectionId, String paperId, String userId);

    Optional<ReadingWorkflowHighlight> findByIdCollectionIdAndIdPaperIdAndIdUserIdAndIdHighlightId(String collectionId, String paperId, String userId, String highlightId);
}

File: services/paper-service/src/main/kotlin/com/se1853_jv/repository/ReadingWorkflowHighlightRepository.kt
package com.se1853_jv.repository

import com.se1853_jv.model.ReadingWorkflowHighlight
import com.se1853_jv.model.ReadingWorkflowHighlightId
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ReadingWorkflowHighlightRepository : JpaRepository<ReadingWorkflowHighlight, ReadingWorkflowHighlightId> {
    
    /**
     * Delete all highlights for a specific reading workflow
     */
    fun deleteByRwCollectionIdAndRwPaperIdAndRwUserId(
        collectionId: String,
        paperId: String,
        userId: String
    )
}







